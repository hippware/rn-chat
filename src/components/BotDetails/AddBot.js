// @flow

import React from 'react';
import {Text, TouchableOpacity, StyleSheet, Image} from 'react-native';
import {colors} from '../../constants';
import {k} from '../../globals';
import {observer} from 'mobx-react/native';

export default observer(({subscribe, unsubscribe, isSubscribed}) => {
  let onPress, buttonStyle, image, text, textStyle;
  if (isSubscribed) {
    onPress = unsubscribe;
    buttonStyle = [styles.addBotButton, {backgroundColor: colors.WHITE, borderColor: colors.GREY}];
    image = require('../../../images/iconCheckBotAdded.png');
    text = 'SAVED';
    textStyle = [styles.addBotText, {color: colors.PURPLE}];
  } else {
    onPress = subscribe;
    buttonStyle = styles.addBotButton;
    image = require('../../../images/saveIcon.png');
    text = 'SAVE';
    textStyle = styles.addBotText;
  }
  return (
    <TouchableOpacity onPress={onPress} style={buttonStyle}>
      <Image source={image} style={styles.addBotIcon} resizeMode='contain' />
      <Text style={textStyle}>
        {text}
      </Text>
    </TouchableOpacity>
  );
});

const styles = StyleSheet.create({
  addBotButton: {
    flexDirection: 'row',
    height: 40 * k,
    marginTop: -20 * k,
    width: 150 * k,
    alignSelf: 'center',
    backgroundColor: colors.PINK,
    borderRadius: 5 * k,
    borderColor: colors.PINK,
    borderWidth: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  addBotText: {
    fontSize: 15,
    letterSpacing: 0.5,
    fontFamily: 'Roboto-Medium',
    color: 'white',
    marginLeft: 5 * k,
  },
});
