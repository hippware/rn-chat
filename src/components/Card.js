import React from 'react'
import { StyleSheet, TouchableOpacity, Animated, Image, View } from 'react-native'
import { k, backgroundColorCardDay, backgroundColorCardNight } from '../globals'
import { observer } from 'mobx-react/native'
import Cell from './Cell'
import autobind from 'autobind-decorator'
import location from '../store/locationStore'

@autobind
@observer
export default class Card extends React.Component {
    constructor (props) {
        super(props)
        this.state = {collapsed: this.props.collapsed, height: this.props.collapsedHeight}
    }

    expand () {
        this.setState({collapsed: false})
    }

    dismiss () {
        this.setState({collapsed: true})
    }

    componentWillReceiveProps (props) {
        if (props.collapsedHeight != this.props.collapsedHeight) {
            this.setState({collapsed: true, height: props.collapsedHeight})
        }
    }

    render () {
        const {style, children, ...props} = this.props
        const isDay = this.props.isDay === undefined ? location.isDay : this.props.isDay
        const backgroundColor = isDay ? backgroundColorCardDay : backgroundColorCardNight
        if (this.props.onPress) {
            return (
                <TouchableOpacity onPress={this.props.onPress}>
                    <View  {...this.props} style={[styles.container, this.props.style]}>
                        <View style={[styles.inner, {backgroundColor}, this.props.innerStyle]}>
                            {React.Children.map(this.props.children, child => child && props ? React.cloneElement(child, props) : child)}
                        </View>
                        {this.props.footer}
                    </View>
                </TouchableOpacity>
            )
        } else {
            return <View  {...this.props} style={[styles.container, this.props.style]}>
                {this.state.collapsed && <View style={{height: this.state.height, overflow: 'hidden'}}>
                    <View style={[styles.inner, {backgroundColor}, this.props.innerStyle]}
                          onLayout={props => this.setState({totalHeight: props.nativeEvent.height})}>
                        {React.Children.map(children, child => child ? (props ? React.cloneElement(child, props) : child) : false)}
                    </View>
                </View>}

                {!this.state.collapsed && <View style={[styles.inner, {backgroundColor}, this.props.innerStyle]}
                                                onLayout={props => this.setState({totalHeight: props.nativeEvent.height})}>
                    {React.Children.map(children, child => child ? (props ? React.cloneElement(child, props) : child) : false)}
                </View>}

                {this.state.collapsed && <View style={{paddingTop: 4,}}><TouchableOpacity onPress={() => this.expand()}>
                    <View style={{alignItems: 'center'}}><Image
                        source={this.props.isDay ? require('../../images/group.png') : require('../../images/groupNight.png')}/></View>
                </TouchableOpacity></View>}

                {this.props.collapsed !== undefined && !this.state.collapsed &&
                <View style={{paddingTop: 4,}}><TouchableOpacity onPress={() => this.dismiss()}>
                    <View style={{alignItems: 'center'}}><Image
                        source={this.props.isDay ? require('../../images/dismiss.png') : require('../../images/dismissNight.png')}/></View>
                </TouchableOpacity></View>}
                {this.props.footer}
            </View>
        }
    }
}

const styles = StyleSheet.create({
    container: {
        paddingRight: 15,
        paddingLeft: 15,
        paddingTop: 13,
        paddingBottom: 10,
    },
    inner: {
        borderColor: 'white',
        borderRadius: 2,
        shadowOffset: {height: 1, width: 0}, shadowRadius: 2, shadowOpacity: 0.12,

    }
})

Card.propTypes = {
    isDay: React.PropTypes.bool,
}
