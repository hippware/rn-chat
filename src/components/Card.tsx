import React from 'react'
import {StyleSheet, TouchableWithoutFeedback, View} from 'react-native'
import {k} from './Global'
import {observer} from 'mobx-react'
import {colors} from '../constants'

type Props = {
  // isDay: boolean,
  style?: any
  innerStyle?: any
  onPress?: any
  footer?: any
  children: any
}

const Card = observer((props: Props) => {
  const {style, children, onPress, footer, innerStyle, ...rest} = props
  const isDay = true
  const backgroundColor = isDay ? colors.backgroundColorCardDay : colors.backgroundColorCardNight
  const inner = (
    <View {...rest} style={[styles.container, style]}>
      <View style={[styles.inner, {backgroundColor}, innerStyle]}>
        {React.Children.map(children, child =>
          child && props ? React.cloneElement(child as any, rest) : child
        )}
      </View>
      {footer}
    </View>
  )
  return onPress ? (
    <TouchableWithoutFeedback onPress={onPress}>{inner}</TouchableWithoutFeedback>
  ) : (
    inner
  )
})

export default Card

const styles = StyleSheet.create({
  container: {
    paddingRight: 15 * k,
    paddingLeft: 15 * k,
    paddingTop: 13 * k,
    paddingBottom: 10 * k,
  },
  inner: {
    borderColor: 'white',
    borderRadius: 2,
    shadowOffset: {height: 1, width: 0},
    shadowRadius: 2,
    shadowOpacity: 0.12,
  },
})
