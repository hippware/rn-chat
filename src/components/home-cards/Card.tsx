import React from 'react'
import {View, StyleSheet, TouchableWithoutFeedback} from 'react-native'
import {IProfile} from 'wocky-client'
import {Avatar} from '../common'
import {k} from '../Global'

type Props = {
  profile?: IProfile
  children: any
  onPress?: () => void
}

const Card = ({profile, children, onPress}: Props) => (
  <TouchableWithoutFeedback onPress={onPress}>
    <View style={styles.card}>
      {profile && (
        <Avatar profile={profile} size={40} style={styles.avatar} hideDot fontFamily="regular" />
      )}
      {children}
    </View>
  </TouchableWithoutFeedback>
)

export default Card

const styles = StyleSheet.create({
  card: {
    backgroundColor: 'white',
    flex: 1,
    flexDirection: 'row',
    marginTop: 20 * k, // leave room for avatar to poke above
    borderRadius: 3,
  },
  avatar: {
    position: 'absolute',
    top: -20 * k,
    left: -14 * k,
    zIndex: 2,
  },
})
