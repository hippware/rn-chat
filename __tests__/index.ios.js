// @flow

// import {NativeModules} from 'react-native';
import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

// @TODO: mocks
// NativeModules.RNCarrierInfo = {isoCountryCode: jest.fn()};

import AddBot from '../src/components/BotDetails/AddBot';
// import BotInfo from '../src/components/BotInfo';
import Card from '../src/components/Card';

describe('Testing jest', () => {
  // test('renders correctly', () => {
  //   console.log('&&& ahahhahahah');
  //   const tree = renderer.create(<Onboarding />).toJSON();
  //   // const tree = renderer.create(<SideMenu />).toJSON();
  //   expect(tree).toMatchSnapshot();
  // });

  // test('Privacy', () => {
  //   const tree = renderer.create(<Privacy />).toJSON();
  //   // const tree = renderer.create(<SideMenu />).toJSON();
  //   expect(tree).toMatchSnapshot();
  // });

  test('AddBot', () => {
    const tree = renderer.create(<AddBot />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Card', () => {
    const tree = renderer.create(<Card />).toJSON();
    // const tree = renderer.create(<SideMenu />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // jest.mock('Linking', () => {
  //   return {
  //     addEventListener: jest.fn(),
  //     removeEventListener: jest.fn(),
  //     openURL: jest.fn(),
  //     canOpenURL: jest.fn(),
  //     getInitialURL: jest.fn(),
  //   };
  // });
  //
  // jest.mock('NetInfo', () => {
  //   return {
  //     isConnected: {
  //       fetch: () => {
  //         return new Promise((accept, resolve) => {
  //           accept(true);
  //         });
  //       },
  //     },
  //   };
  // });
  //
  // jest.mock('PushNotificationIOS', () => {
  //   return {
  //     getInitialNotification: () => {
  //       return new Promise((accept, resolve) => {
  //         accept(true);
  //       });
  //     },
  //   };
  // });
  // //
  // jest.mock(
  //   'NativeEventEmitter',
  //   () =>
  //     class NEE {
  //       constructor() {}
  //       addListener = jest.fn();
  //     }
  // );
  //
  // test('BotInfo', () => {
  //   const tree = renderer.create(<BotInfo />).toJSON();
  //   expect(tree).toMatchSnapshot();
  // });
});
