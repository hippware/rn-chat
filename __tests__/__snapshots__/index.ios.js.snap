exports[`Testing jest AddBot 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "rgb(253,95,108)",
      "borderColor": "rgb(253,95,108)",
      "borderRadius": 5,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 40,
      "justifyContent": "center",
      "marginTop": -20,
      "opacity": 1,
      "width": 150,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}>
  <Image
    resizeMode="contain"
    source={1}
    style={undefined} />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontFamily": "Roboto-Medium",
        "fontSize": 15,
        "letterSpacing": 0.5,
        "marginLeft": 5,
      }
    }>
    SAVE
  </Text>
</View>
`;

exports[`Testing jest Card 1`] = `
<View
  style={
    Array [
      Object {
        "paddingBottom": 20,
        "paddingLeft": 30,
        "paddingRight": 30,
        "paddingTop": 26,
      },
      undefined,
    ]
  }>
  <View
    style={
      Array [
        Object {
          "borderColor": "white",
          "borderRadius": 2,
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.12,
          "shadowRadius": 2,
        },
        Object {
          "backgroundColor": "rgba(255,255,255,1)",
        },
        undefined,
      ]
    } />
</View>
`;
